src/hooks/useFetchUnFollow.tsx

```diff
@@ -0,0 +1,52 @@
+import { AUTH_API } from "@/apis/Api"
+import AlertModal from "@/components/Modal/components/AlertModal/AlertModal"
+import useModal from "@/components/Modal/hooks/useModal"
+import { POST_DETAIL_ERROR_MESSAGE } from "@/constants/errorMessage"
+import useAuthUserStore from "@/stores/useAuthUserStore"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+
+const FETCH_MUTATION_UN_FOLLOW_KEY =
+  "IT_IS_FETCH_MUTATION_KEY_781521784628175281"
+
+const useFetchUnFollow = () => {
+  const { updateUser } = useAuthUserStore()
+  const queryClient = useQueryClient()
+  const fetchUnFollowMutate = useMutation({
+    mutationKey: [FETCH_MUTATION_UN_FOLLOW_KEY],
+    mutationFn: fetchUnFollow,
+    onSuccess: () => {
+      queryClient.refetchQueries()
+
+      AUTH_API.get("/auth-user")
+        .then((res) => res.data)
+        .then((data) => updateUser(data))
+    },
+    onError: () => {
+      showModal()
+    },
+  })
+  const { isShowModal, showModal, closeModal } = useModal()
+
+  const UnFollowErrorAlertModal = isShowModal && (
+    <AlertModal
+      isShow={isShowModal}
+      onClose={closeModal}
+      alertMessage={POST_DETAIL_ERROR_MESSAGE.USER.UN_FOLLOW}
+    />
+  )
+
+  return { fetchUnFollowMutate, UnFollowErrorAlertModal }
+}
+
+export default useFetchUnFollow
+
+const fetchUnFollow = (id: string) => {
+  const response = AUTH_API.delete("/follow/delete", {
+    data: { id },
+  })
+    .then((res) => res)
+    .catch((e) => {
+      throw Error(e)
+    })
+  return response
```

Comment:
움.. `useFetchFollow.tsx` 와 비교했을 때
```suggestion
const fetchUnFollow = async (id: string) => {
  const response = await AUTH_API.delete("/follow/delete", {
    userId: id,
  })
    .then((res) => res.data)
    .catch((e) => {
      throw Error(e)
    })
  return response
```
이렇게 하는게 맞을 것 같은데 제가 놓치고 있는 부분이 있다면 말씀해주시면 감사하겠습니다!

Reply:
> 움.. `useFetchFollow.tsx` 와 비교했을 때
> 
> 이렇게 하는게 맞을 것 같은데 제가 놓치고 있는 부분이 있다면 말씀해주시면 감사하겠습니다!

delete는 API 호출시 저렇게 data안에 값을 넣어야 하더라구요.. 그래서 지금 하고 있는 방식이 아마 맞을 겁니다!

Reply:
맞슴돵 그 post put get delete 에 따라 data전달 방식이 다르더라구요!